packages/client/src/core/storage/ScopedAsyncStorage.ts

```diff
@@ -1,6 +1,9 @@
-import AsyncStorage from '@react-native-async-storage/async-storage';
+import AsyncStorage_, { AsyncStorageStatic } from '@react-native-async-storage/async-storage';
 
-import { KeyValueStorage, Scope, ScopedStorage } from './types';
+import { KeyValueStorage, Scope, ScopedStorage } from './types.js';
+
+// AsyncStorage type not correctly inferred by compiler so we have to cast it manually
+const AsyncStorage = AsyncStorage_ as unknown as AsyncStorageStatic;
```

Comment:
This was the one weird issue I ran into when migrating. Seems like the type isn't being inferred correctly from the default export